src/components/App.js

```diff
@@ -0,0 +1,31 @@
+import React, { Component } from 'react'
+import Display from './Display'
+import ButtonPanel from './ButtonPanel'
+import calculate from '../logic/calculate'
+
+class App extends Component {
+  constructor (props) {
```

Comment:
All well and good here. Though I would recommend we just use the new syntax for classes. ie.:

```javascript
class App extends React.Component {
  state = {
    // same
  }

  handleClick = buttonName => this.setState(prevState => calculate(prevState))
}
```